pkg/controller/che/finalizer.go

```diff
@@ -20,14 +21,15 @@ func (r *ReconcileChe) ReconcileFinalizer(instance *orgv1.CheCluster) (err error
 			oAuthClientName := instance.Spec.Auth.OauthClientName
 			logrus.Infof("Custom resource %s is being deleted. Deleting oAuthClient %s first", instance.Name, oAuthClientName)
 			oAuthClient, err := r.GetOAuthClient(oAuthClientName)
-			if err != nil {
+			if err == nil {
+				if err := r.client.Delete(context.TODO(), oAuthClient); err != nil {
+					logrus.Errorf("Failed to delete %s oAuthClient: %s", oAuthClientName, err)
+					return err
+				}
+			} else if !errors.IsNotFound(err) {
```

Comment:
What if `err != nil` and `errors.IsNotFound(err)`?

Reply:
Maybe you should

```go
if errors.IsNotFound(err) {
  ...
} else {
  ...
}

Reply:
if `err != nil` and `errors.IsNotFound(err)`, we:
- ignore the error
- don't try to delete the `OAuthClient` that doesn't exist
- continue normally (=> delete the finalizer and go on) 

in brief:
```go
if errors.IsNotFound(err) {
  // do nothing with OAuthClient and continue
}
```